src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -679,6 +914,7 @@ def __init__(self, config):
         self.model_parallel = False
         self.device_map = None
         self.gradient_checkpointing = False
+        self._use_flash_attention_2 = config._attn_implementation == "flash_attention_2"
```

Comment:
```suggestion
        self._attn_implementation = config._attn_implementation 
```
once we'll have sdpa as well this shall be re-used